test/functional/tests/io/trim/test_trim_eviction.py

```diff
@@ -0,0 +1,88 @@
+#
+# Copyright(c) 2020 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause-Clear
+#
+
+import os
+import pytest
+from api.cas import casadm
+from api.cas.cache_config import CacheMode, CacheLineSize, CleaningPolicy
+from core.test_run import TestRun
+from storage_devices.disk import DiskTypeSet, DiskType, DiskTypeLowerThan
+from test_tools import fs_utils
+from test_tools.ddrescue import Ddrescue
+from test_tools.disk_utils import Filesystem
+from test_utils import os_utils
+from test_utils.os_utils import Udev
+from test_utils.size import Size, Unit
+
+
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand, DiskType.sata]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+@pytest.mark.parametrize("cache_mode", CacheMode)
+@pytest.mark.parametrize("cache_line_size", CacheLineSize)
+@pytest.mark.parametrize("filesystem", [Filesystem.ext4, Filesystem.xfs])
+@pytest.mark.parametrize("cleaning", [CleaningPolicy.alru, CleaningPolicy.nop])
+def test_trim_eviction(cache_mode, cache_line_size, filesystem, cleaning):
+    """
+        title: Test verifying if trim requests do not cause eviction on CAS device.
+        description: |
+          When trim requests enabled and files are being added and removed from CAS device,
+          there is no eviction (no reads from cache).
+        pass_criteria:
+          - Reads from cache device are the same before and after removing test file.
+    """
+    mount_point = "/mnt"
+    test_file_path = os.path.join(mount_point, "test_file")
+
+    with TestRun.step("Prepare devices."):
+        cache_disk = TestRun.disks['cache']
+        cache_disk.create_partitions([Size(1, Unit.GibiByte)])
+        cache_dev = cache_disk.partitions[0]
+
+        core_disk = TestRun.disks['core']
+        core_disk.create_partitions([Size(1, Unit.GibiByte)])
+        core_dev = core_disk.partitions[0]
+
+        with TestRun.step("Start cache on device supporting trim and add core."):
```

Comment:
Indent for this and following steps should be the same as for the "Prepare devices" step.